src/sage/schemes/projective/cohomology.py

```diff
@@ -0,0 +1,500 @@
+                        assert(sum(u) == -H0.shifts[j])
+                        if any(e >= 0 for e in u):
+                            continue
+                        k = H0.summands_index[j] + basis.index(u)
+                        image[k] += c
+                A.append(vector(K, image))
+
+        return H1.vector_space.hom(A, codomain=H0.vector_space, side='right')
+
+    def H(self, q):
+        r"""
+        Return the `q`-th cohomology group `H^q(\tilde M)`.
+
+        INPUT:
+
+        - ``q`` -- non-negative integer
+
+        EXAMPLES::
+
+            sage: P2.<x,y,z> = ProjectiveSpace(QQ, 2)
+            sage: X = P2.subscheme([x^4 + y^4 + z^4])
+            sage: c = X.structure_sheaf()._cohomology
+            sage: c.H(1)
+            Vector space quotient V/W of dimension 3 over Rational Field where
+            V: Vector space of degree 3 and dimension 3 over Rational Field
+            Basis matrix:
+            [1 0 0]
+            [0 1 0]
+            [0 0 1]
+            W: Vector space of degree 3 and dimension 0 over Rational Field
+            Basis matrix:
+            []
+        """
+        r = self.projective_space_dimension
+        if q == r:
+            return self.cohomology_group_top(0).vector_space.quotient(self.differential_top(1).image())
+        elif 1 <= q and q < r:
+            return self.differential_top(r - q).kernel().quotient(self.differential_top(r - q + 1).image())
+        elif q == 0:
+            # 0th cohomology group of (global sections) is not realized in
+            # terms of cohomology groups and differential maps of twisted
+            # structure sheaves of the projective space.
+            return VectorSpace(self.coefficient_field, self.h(0))
+        elif q > r:
+            return VectorSpace(self.coefficient_field, 0)
+        raise IndexError('index must be non-negative')
+
+    def h(self, q):
+        r"""
+        Return the dimension `h^q(\tilde M)` of the `q`-th cohomology group `H^q(\tilde M)`.
+
+        INPUT:
+
+        - ``q`` -- non-negative integer
+
+        EXAMPLES::
+
+            sage: P2.<x,y,z> = ProjectiveSpace(QQ, 2)
+            sage: X = P2.subscheme([x^4 + y^4 + z^4])
+            sage: c = X.structure_sheaf()._cohomology
+            sage: c.h(1)
+            3
+        """
+        r = self.projective_space_dimension
+        if q == r:
+            return self.cohomology_group_top(0).rank - self.differential_top(1).rank()
+        elif 1 <= q and q < r:
+            return self.differential_top(r - q).kernel().dimension() - self.differential_top(r - q + 1).rank()
+        elif q == 0:
+            a = (self.cohomology_group_bottom(0).rank - self.cohomology_group_top(r + 1).rank
+                 + self.cohomology_group_top(r).rank)
+            b = self.differential_bottom(1).rank() + self.differential_top(r).rank()
+            return a - b
+        elif q > r:
+            return 0
+        raise IndexError('index must be non-negative')
```

Comment:
```suggestion
        return ZZ.zero()
```
Same comment. Also, let's try to make sure Python `int`'s don't get returned.

Reply:
Right. I will use `Integer(0)`.